src/transaction/txParser.ts

```diff
@@ -204,7 +264,11 @@ const parseUnsignedTx = ([txBody, meta]: _UnsignedTxDecoded): _UnsignedTxParsed
     txBody.get(TxBodyKeys.WITHDRAWALS) || new Map(),
   )
   const metaDataHash = txBody.get(TxBodyKeys.META_DATA_HASH) as Buffer
-  return {
+  const validityIntervalStartRaw = txBody.get(TxBodyKeys.VALIDITY_INTERVAL_START)
+  const validityIntervalStart = validityIntervalStartRaw
```

Comment:
Maybe we could just call `parseValidityInterval(txBody.get(...))` as we are doing elsewhere and have the function return undefined if the key is undefined, if its not, it should check if its valid.

Also we should parse the ttl in the same way, returning undefined if its the key is empty. Similary we should create a function for parsing `fee` which would do the same basiccaly, but would throw an error if its empty.

Reply:
I did that for now. I think as we are adding more optional fields, it would be nice to have some wrapper optional function somewhere.